src/cpp/src/logit_processor.hpp

```diff
@@ -174,14 +167,14 @@ class EOSPenaltyTransform : public ILogitTransformer {
     EOSPenaltyTransform(size_t eos_token_id, size_t min_generated_tokens) : 
         m_eos_token_id(eos_token_id), m_applicable_tensor_len(min_generated_tokens) {}
 
-    std::vector<Token> apply(const std::vector<Token>& input_logits) {
-        std::vector<Token> output(input_logits.begin(), input_logits.end());
-        for (auto& token_id : output) {
+    void apply(std::vector<Token>& logits) override {
+        for (auto& token_id : logits) {
+            // TODO: Is that right? m_index seems to be only a location of the token that logit is bound to
+            // while here we compare that index to token id
             if (token_id.m_index == m_eos_token_id) {
```

Comment:
Could you please fix it? Looks you are right. Please check `set_eos_token in the generation config`

Reply:
Actually I think I was wrong here. At some point I've seen that logits size is smaller than vocab size and hence I had a suspicion that this might not work as expected, but I've check that the size is matching so such index to id comparison seems to be valid here.